transport_nantes/topicblog/views.py

```diff
@@ -222,10 +226,25 @@ def get_context_data(self, **kwargs):
         slug = kwargs.get('the_slug', '')
         tb_object = get_object_or_404(self.model, id=pk_id, slug=slug)
 
-        # We set the template in the model.
-        self.template_name = tb_object.template_name
+        # We check that the object's template is valid and configured, or
+        # fallback to a default template.
+        # Doing so we make sure that if a template is retired, we can keep
+        # serving the content without raising a 500 error.
+        if tb_object.template_name in tb_object.template_config:
+            self.template_name = tb_object.template_name
+        else:
+            logger.info(
+                f"Current template ({tb_object.template_name}) is not configured."
```

Comment:
I'd say this is an error, no?

It's good we catch it, but we should get a rollbar alert if it ever happens.

Reply:
After all, one should assume that changing the template could do really horrible things to what is displayed: different assumptions about what fields are displayed and in what juxtaposition, etc.